### ISSUE TITLE
SDDM doesn't support `passwd -de`

### ISSUE THREAD

**Issue by Mikaela** [sddm/sddm#472]
As I don't want to know passwords of my family members, I reset them with `passwd -de` after reinstalling OS or they have forgotten them.

With LightDM this prompts a message that root has enforced the user to change password and asks for new password, confirmation for it and then the user can login using that password.

SDDM only says `Login failed`.

From the passwd flags, d removes the current password and e marks the password as expired requiring new one during next login.

* OS: Antergos
* SDDM version: 0.11.0-3

**Comment by plfiorini** [comment]
Our PAM library *should* have everything to detect this situation.
Most likely a missing feature on the themes and/or propagation of the PAM thingy.
Definitely something we want to have.

**Comment by Mikaela** [comment]
I am using the maldives theme and in case it helps my sddm.conf is:

```
[Autologin]
# Autologin again on session exit
Relogin=false

# Autologin session
Session=

# Autologin user
User=


[General]
# Halt command
HaltCommand=/usr/bin/systemctl poweroff

# Initial NumLock state
# Valid values: on|off|none
# If property is set to none, numlock won't be changed
Numlock=on

# Reboot command
RebootCommand=/usr/bin/systemctl reboot


[Theme]
# Current theme name
Current=maldives

# Cursor theme
CursorTheme=

# Face icon directory
# The files should be in username.face.icon format
FacesDir=/usr/share/sddm/faces

# Theme directory path
ThemeDir=/usr/share/sddm/themes


[Users]
# Default $PATH
DefaultPath=/bin:/usr/bin:/usr/local/bin

# Hidden shells
# Users with these shells as their default won't be listed
HideShells=

# Hidden users
HideUsers=

# Maximum user id for displayed users
MaximumUid=65000

# Minimum user id for displayed users
MinimumUid=1000

# Remember the session of the last successfully logged in user
RememberLastSession=true

# Remember the last successfully logged in user
RememberLastUser=true


[XDisplay]
# Xsetup script path
# A script to execute when starting the display server
DisplayCommand=/usr/share/sddm/scripts/Xsetup

# Minimum VT
# The lowest virtual terminal number that will be used.
MinimumVT=1

# X server path
ServerPath=/usr/bin/X

# Xsession script path
# A script to execute when starting the desktop session
SessionCommand=/usr/share/sddm/scripts/Xsession

# Session description directory
SessionDir=/usr/share/xsessions

# Xauth path
XauthPath=/usr/bin/xauth
```

**Comment by Mikaela** [comment]
Sorry, accidental close.

**Comment by bobOnGitHub** [comment]
Workaround: get the user to ctrl+alt+F\<n\> to get a terminal login prompt and do the password change there, then logout (exit) in that terminal and ctrl-alt-F\<n\> back to the sddm for a login to KDE. Should work everywhere but tested on Alma Linux 8.6.

**Comment by severgun** [comment]
10 years dealbreaker without progress

**Comment by h908714124** [comment]
We had this problem and switch to gdm, as a temp workaround (ugly, I know, because that has gtk dependencies, but works) until this is resolved.
Lightdm did not work for us, system didn't appear to boot anymore, we could not (quickly) figure out why.

update: lightdm works too, we had invalid lines in `/etc/environment`.

**Comment by YamiYukiSenpai** [comment]
Ran into same issue, but with an Active Directory account

```
sudo systemctl status sddm
● sddm.service - Simple Desktop Display Manager
     Loaded: loaded (/usr/lib/systemd/system/sddm.service; enabled; preset: enabled)
     Active: active (running) since Mon 2024-11-25 09:03:13 EST; 33s ago
       Docs: man:sddm(1)
             man:sddm.conf(5)
    Process: 2327 ExecStartPre=/bin/sh -c [ "$(cat /etc/X11/default-display-manager 2>/dev/null)" = "/usr/bin/sddm" ] (code=exited, status=0/SUCCESS)
   Main PID: 2330 (sddm)
      Tasks: 19 (limit: 18753)
     Memory: 42.8M (peak: 43.3M)
        CPU: 873ms
     CGroup: /system.slice/sddm.service
             ├─2330 /usr/bin/sddm
             ├─2350 /usr/lib/xorg/Xorg -nolisten tcp -background none -seat seat0 vt2 -auth /run/sddm/xauth_AtAHja -noreset -displayfd 16
             └─2492 /usr/lib/x86_64-linux-gnu/sddm/sddm-helper --socket /tmp/sddm-auth-6a0e1b27-ec96-4ead-9583-ed9435b42f8e --id 1 --start /usr/bin/startplasma-x11 --user pyor>

Nov 25 09:03:23 sddm-helper[2492]: pam_sss(sddm:account): User info message: Password expired. Change your password now.
Nov 25 09:03:23 sddm-helper[2492]: [PAM] Preparing to converse...
Nov 25 09:03:23 sddm-helper[2492]: [PAM] Conversation with 1 messages
Nov 25 09:03:23 sddm[2330]: Authentication information: SDDM::Auth::INFO_UNKNOWN "Password expired. Change your password now."
Nov 25 09:03:23 sddm-helper[2492]: pam_unix(sddm:chauthtok): user "pyorgova" does not exist in /etc/passwd
Nov 25 09:03:23 sddm-helper[2492]: [PAM] Preparing to converse...
Nov 25 09:03:23 sddm-helper[2492]: [PAM] Conversation with 1 messages
Nov 25 09:03:30 sddm[2330]: Message received from greeter: Login
Nov 25 09:03:30 sddm[2330]: Reading from "/usr/share/xsessions/plasma.desktop"
Nov 25 09:03:30 sddm[2330]: Existing authentication ongoing, aborting
```

Is there a solution already?

**Comment by MaKaNu** [comment]
@jbruechert what is the progress on this branch https://github.com/sddm/sddm/compare/expired-password?

It is hard to determine what was developed on the branch with such amount of changes. (Is this how this repo working?)

**Comment by jbruechert** [comment]
No, the reason that this commit is not merged is that it is too large to review most likely.
The only thing I did is rebase the branch once more, I don't plan on working on this any further, so there is no progress.

**Comment by MaKaNu** [comment]
@jbruechert Just found the PR #776 of @hoehnth. Sad to see that such feature did not get the love it needed...

**Comment by jbruechert** [comment]
I'm repeating myself here, but after that PR the venn diagram of people who want this feature and people who actually do the work is two separate circles.
I can understand that nobody felt confident to merge a single commit hugely changing the core of SDDM.
The refactoring needs to happen first in separate commits, and then this feature can be implemented.

**Comment by hoehnth** [comment]
The PR was provided with separate commits initially. But after request to squash all commits, which I did, then discussion came up again to split it up in separate commits, which seems to me there was no serious interest to merge it. Have lost interest to do any more work on it.

**Comment by MaKaNu** [comment]
@jbruechert I would like to contribute. I am not deeply into the core of the project, but know a bit about the qml frontend. I might take a look into it.

@hoehnth is the history on your fork still intact?
